metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreUtils.java

```diff
@@ -154,7 +154,7 @@ static public Deserializer getDeserializer(Configuration conf,
     ObjectInspector oi = deserializer.getObjectInspector();
     String[] names = tableName.split("\\.");
     String last_name = names[names.length - 1];
-    for (int i = 1; i < names.length; i++) {
+    for (int i = 2; i < names.length; i++) {
```

Comment:
I'm not sure what's happening here ; but this =1  / =2 looks wierd

Reply:
This used to split e.g. `mytable.field1`, however now we're always passing `mydb.mytable.field1`. I can leave a comment to clarify this.